src/main/java/com/linkedin/kmf/common/DefaultTopicSchema.java

```diff
@@ -15,17 +15,17 @@
 
 public class DefaultTopicSchema {
 
-  public static final Field TOPIC_FIELD = new Field("topic", Schema.create(Schema.Type.STRING), null, null);
+  static final Field TOPIC_FIELD = new Field("topic", Schema.create(Schema.Type.STRING), null, null);
```

Comment:
Nit: Reducing these unnecessary access scope (surface area of a class) is definitely a good thing to do in general. But on the other hand, this class is defined in the `common` package and it implies that this class is likely or supposed to be used in a shared way. So making some of the static functions package private just because they are not used at this point seems a little unnecessary for me. It's just my opinion (you don't have to change anything). But I'd like to know your thoughts about it.

Reply:
You've raised a reasonable point. Decreasing the surface area of a class is a sound thing to perform. Restricting the avenues which classes can interact with this one makes it simpler to render internal modifications that are not visible to those other classes. We need to expose only to **bare minimum** which is required to fulfill these requirements.

`DefaultTopicSchema` is in fact defined within the `common` package. However public `surface area` for a class effectively defines its `API`, and we can expose it back to public if it should be accessible to wider scope in the future. 

If variables are exposed via methods, the logistics of how they work can be modified later.

When exposing  a field, some other class outside of control and maybe outside of **knowledge** beings referencing such field -> leading to being stuck with exposing that field for the rest of time. This might lead up to breaking `backwards-compatibility`.

In general, not  "all classes" should have access them. In pursuit of **maintainability** of source code, we shouldn't necessarily go for the most convenient to write in the first place.